tests/python/gaia-ui-tests/gaiatest/apps/homescreen/regions/bookmark_menu.py

```diff
@@ -14,35 +14,27 @@ class BookmarkMenu(Base):
     name = "Bookmark"
     manifest_url = '{}bookmark{}/manifest.webapp'.format(Base.DEFAULT_PROTOCOL,Base.DEFAULT_APP_HOSTNAME)
 
-    # System app - add bookmark to homescreen dialog
-    _add_bookmark_to_home_screen_frame_locator = (By.CSS_SELECTOR, 'iframe[src^="app://bookmark"][src$="save.html"]')
-    _add_bookmark_to_home_screen_dialog_button_locator = (By.ID, 'done-button')
-    _bookmark_title_input_locator = (By.ID, 'bookmark-title')
+    _pin_site_to_home_screen = (By.CSS_SELECTOR, 'button[data-action="pin-site"]')
+    _pin_page_to_home_screen = (By.CSS_SELECTOR, 'button[data-action="pin-page"]')
 
     def __init__(self, marionette):
         Base.__init__(self, marionette)
         self.wait_to_be_displayed()
         self.apps.switch_to_displayed_app()
 
-    def tap_add_bookmark_to_home_screen_dialog_button(self):
+
+    def tap_pin_site_to_home_screen(self):
         element = Wait(self.marionette).until(expected.element_present(
-            *self._add_bookmark_to_home_screen_dialog_button_locator))
+            *self._pin_site_to_home_screen))
         Wait(self.marionette).until(expected.element_displayed(element))
         # This sleep is necessary for the button to react to the tap call
         time.sleep(0.2)
         self.tap_element_from_system_app(element)
 
-        # Wait for the Add to bookmark frame to be dismissed
-        self.wait_to_not_be_displayed()
-        self.apps.switch_to_displayed_app()
-
-    def type_bookmark_title(self, value):
-        element = self.marionette.find_element(
-            *self._bookmark_title_input_locator)
-
-        # Wait for the default value to load into the input field
-        Wait(self.marionette).until(lambda m: element.get_attribute('value') != '')
-        element.clear()
-
-        self.keyboard.send(value)
-        self.keyboard.dismiss()
+    def tap_pin_page_to_home_screen(self):
+        element = Wait(self.marionette).until(expected.element_present(
+            *self._pin_page_to_home_screen))
+        Wait(self.marionette).until(expected.element_displayed(element))
+        # This sleep is necessary for the button to react to the tap call
+        time.sleep(0.2)
```

Comment:
I wonder if we can get rid of this wait by making sure the button is enabled: http://marionette-client.readthedocs.org/en/latest/reference.html?highlight=html%20element#marionette_driver.marionette.HTMLElement.is_enabled